relengapi/blueprints/slaveloan/__init__.py

```diff
@@ -0,0 +1,233 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import flask_login
+import logging
+import sqlalchemy as sa
+import wsme
+
+from sqlalchemy import asc
+
+from flask import Blueprint
+from flask import g
+from flask import render_template
+from relengapi import apimethod
+from relengapi import p
+from relengapi.blueprints.slaveloan import task_groups
+from relengapi.blueprints.slaveloan.slave_mappings import slave_patterns
+from relengapi.blueprints.slaveloan.slave_mappings import slave_to_slavetype
+from relengapi.util import tz
+from werkzeug.exceptions import BadRequest
+from werkzeug.exceptions import InternalServerError
+
+from relengapi.blueprints.slaveloan import rest
+from relengapi.blueprints.slaveloan.model import History
+from relengapi.blueprints.slaveloan.model import Humans
+from relengapi.blueprints.slaveloan.model import Loans
+from relengapi.blueprints.slaveloan.model import Machines
+
+logger = logging.getLogger(__name__)
+
+bp = Blueprint('slaveloan', __name__,
+               template_folder='templates',
+               static_folder='static')
+
+p.slaveloan.admin.doc("Administer Slaveloans for all users")
+
+##################
+#  RESTful APIs  #
+##################
+
+
+@bp.route('/loans/')
+@p.slaveloan.admin.require()
+@apimethod([rest.Loan])
+def get_loans():
+    "Get the list of all `active` loans you can see"
+    # XXX: Use permissions to filter if not an admin
+    session = g.db.session('relengapi')
+    loans = session.query(Loans).filter(Loans.machine_id.isnot(None))
```

Comment:
You can also write this as `Loans.query.filter(...)`, with no need to fetch a session.